packages/amplify-category-analytics/src/migrations/in-app-messaging-migration.ts

```diff
@@ -28,14 +29,15 @@ import {
 export const inAppMessagingMigrationCheck = async (context: $TSContext): Promise<void> => {
   const projectBackendDirPath = pathManager.getBackendDirPath();
   const resources = getAnalyticsResources(context);
+  const hasPinpointResource = pinpointHelper.hasResource(context);
 
-  if (resources.length > 0 && !pinpointHasInAppMessagingPolicy(context)) {
+  if (resources.length > 0 && hasPinpointResource && !pinpointHasInAppMessagingPolicy(context)) {
     const amplifyMeta = stateManager.getMeta();
     const analytics = amplifyMeta[AmplifyCategories.ANALYTICS] || {};
     Object.keys(analytics).forEach((resourceName) => {
       const analyticsResourcePath = path.join(projectBackendDirPath, AmplifyCategories.ANALYTICS, resourceName);
       const templateFilePath = path.join(analyticsResourcePath, 'pinpoint-cloudformation-template.json');
-      const cfn = JSONUtilities.readJson(templateFilePath);
+      const cfn = JSONUtilities.readJson(templateFilePath, { throwIfNotExist: false });
```

Comment:
`migratePinpointCFN` will fail if cfn is undefined.

Reply:
removed this change as it was a left over before I added the `hasPinpointResource` check